android/src/main/java/com/opentokreactnative/OTSessionManager.java

```diff
@@ -167,28 +172,33 @@ public void initPublisher(String publisherId, ReadableMap properties, Callback c
     }
 
     @ReactMethod
-    public void publish(String publisherId, Callback callback) {
-
-        Session mSession = sharedState.getSession();
-        ConcurrentHashMap<String, Publisher> mPublishers = sharedState.getPublishers();
-        Publisher mPublisher = mPublishers.get(publisherId);
-        if (mSession != null && mPublisher != null) {
-            mSession.publish(mPublisher);
-            callback.invoke();
+    public void publish(String sessionId, String publisherId, Callback callback) {
+        ConcurrentHashMap<String, Session> mSessions = sharedState.getSessions();
+        Session mSession = mSessions.get(sessionId);
+        if (mSession != null) {
+            ConcurrentHashMap<String, Publisher> mPublishers = sharedState.getPublishers();
+            Publisher mPublisher = mPublishers.get(publisherId);
+            if (mPublisher != null) {
```

Comment:
Shouldn't we check if the mPublisher has already a session attached? Or it can publish the same stream in different sessions?

Reply:
Since the OTPublisher is within a OTSession, we have always that attached. When you connect to a session, once you are connected it will try to publish. In the weird case you want to call .publish() again while you are already publishing, you will get an error from Opentok that the app can handle.